### PR TITLE
Misc Missile and CreateEffectWarhead tweaks

### LINKED ISSUE #10350: Missiles no longer trigger splash reliably when hitting water
Observed on RA destroyer and rocket soldier when shooting at empty water. More often than not, an explosion is shown instead.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29495037-missiles-no-longer-trigger-splash-reliably-when-hitting-water?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by reaperrr**
- Fixes #10350 by using `airMargin` in CEWH water check
- unhardcodes `Missile` freefall gravity
- Make `CreateEffectWarhead` check for `HitShape.DistanceFromEdge` instead of `.OuterRadius` (to match `SpreadDamageWarhead`)

### CODE REVIEW

File: OpenRA.Mods.Common/Effects/Missile.cs [hunk 1]
```diff
@@ -76,6 +76,9 @@ public class MissileInfo : IProjectileInfo
 		[Desc("Vertical rate of turn.")]
 		public readonly int VerticalRateOfTurn = 6;
 
+		[Desc("Gravity applied while in free fall.")]
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Effects/Missile.cs]: Could you please add that the negative of this value is used internally?
Or better yet make the default `-10` and use whatever value the modder actually put in yaml.

Comment (reaperrr) [inline OpenRA.Mods.Common/Effects/Missile.cs]: I felt that negative values are unintuitive for most 'end-users'/modders (negative gravity means "falling" towards the sky in my book).

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Effects/Missile.cs]: Okay, that is a fair point.
I'd rather we come up with a better name (something implying a force acting on the missile) and use the values given to us.

But if you're against that and others are, no problem, just please update the description.

Comment (pchote) [inline OpenRA.Mods.Common/Effects/Missile.cs]: I agree with @reaperrr.  This claims to be gravity, not an acceleration vector (which would indeed need to be negative).

Comment (reaperrr) [inline OpenRA.Mods.Common/Effects/Missile.cs]: > Could you please add that the negative of this value is used internally?

Personally, I think modders don't need to know how this works internally. In fact I'm afraid adding that to the description might just cause unnecessary confusion.

Comment (GraionDilach) [inline OpenRA.Mods.Common/Effects/Missile.cs]: Siding with @reaperrr here. Gravity was falling in TS/RA2 as well. The description is fine.

**Comment by RoosterDragon** [comment]
:+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/048fac368fdb63b65615fd0dae2de772398e5a61...34b61a0a82735428140a6a447dac782569fdb24e)

**Pull request merged**